src/reusedBrowser.ts

```diff
@@ -188,27 +196,29 @@ test('test', async ({ page }) => {
       return;
     if (!this._checkVersion(config, 'Show & reuse browser'))
       return;
-    await this.startIfNeeded(config);
+    await this._startBackendIfNeeded(config);
     this._backend?.setAutoClose({ enabled: false });
   }
 
   async didRunTests(config: TestConfig) {
     this._backend?.setAutoClose({ enabled: true });
   }
 
-  private async _cleanupModes() {
+  private async _reset() {
     // This won't wait for setMode(none).
-    if (this._selectorExplorerBox) {
-      this._selectorExplorerBox.dispose();
-      this._selectorExplorerBox = undefined;
-    }
+    this._updateOrCancelInspecting?.({ cancel: true });
+    this._updateOrCancelInspecting = undefined;
+    this._cancelRecording?.();
```

Comment:
This will call `_reset()` again in line 166, but I guess it's fine?

Reply:
Yeah, should be microtasked...